core/src/main/java/com/theagilemonkeys/ellmental/core/schema/Embedding.java

```diff
@@ -1,14 +1,22 @@
 package com.theagilemonkeys.ellmental.core.schema;
 
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 /**
  * Embeddings represent a point in the embeddings space, representing the semantics of a given text.
  */
-public record Embedding(
-        UUID id,
-        List<Double> vector,
-        Map<String, String> metadata
-) {}
+@Data
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class Embedding {
+    public final UUID id;
+    public final List<Double> vector;
+    public final Map<String, String> metadata;
+    public Double score;
```

Comment:
should we use private properties and the getters lombok gives us with @Data?

Reply:
That would make the usage of the class more verbose, and this class is just a data transport object with no logic beyond holding the data. What do we gain if we use the getters instead? Immutability maybe?

Reply:
Yeah, that's the only con I think. Just stayed it to stick to the "classic java" code style, but it's not a real concern or anything 😛